stats/statsd.go

```diff
@@ -200,7 +211,13 @@ func (s *statsdStats) internalNewTaggedStat(name, statType string, tags Tags, sa
 			taggedClient = &statsdClient{samplingRate: samplingRate, tags: tagVals}
 			if s.state.connEstablished {
 				taggedClient.statsdMu.Lock()
```

Comment:
There is no need to lock here given that we just created the client and it's not a key on any of the maps, right?